lib/revocation.js

```diff
@@ -0,0 +1,135 @@
+var assert = require('assert')
+    , async = require('async')
+    , aws = require('aws-sdk')
+    , LRU = require('lru-cache')
+    ;
+
+var dynamo;
+var lru; // LRU cache of token revocation information
+
+exports.init = function (options) {
```

Comment:
Does this get called multiple times from different places in the same node process (with a *different* set of options)?

If not, why not just throw `dynamo` and `lru` into the closure and take out the re-initialization guards? Alternatively, store the fully-initialized service in a module-scoped var that closes around the lru and dynamo instances.